plugins/plugin-java/che-plugin-java-ext-jdt/che-jdt-ext-machine/src/main/java/org/eclipse/che/jdt/rest/JavaNavigationService.java

```diff
@@ -70,6 +74,24 @@ public OpenDeclarationDescriptor findDeclaration(@QueryParam("projectpath") Stri
         return navigation.getProjectDependecyJars(project);
     }
 
+    @GET
+    @Path("find-type")
+    @Produces("application/json")
+    public OpenDeclarationDescriptor findType(@QueryParam("projectpath") String projectPath,
+                                              @QueryParam("fqn") String fqn) throws JavaModelException, ApiException {
+        requireNonNullAndNonEmpty(projectPath, "Project path required");
```

Comment:
I like more this style
https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/recipe/RecipeService.java#L98-L100